modules/highgui/src/window_w32.cpp

```diff
@@ -1874,21 +1874,41 @@ static void showSaveDialog(CvWindow* window)
     ofn.lStructSize = sizeof(ofn);
 #endif
     ofn.hwndOwner = window->hwnd;
+#ifdef HAVE_PNG   
     ofn.lpstrFilter = "Portable Network Graphics files (*.png)\0*.png\0"
+#else
+    ofn.lpstrFilter = "Windows bitmap (*.bmp;*.dib)\0*.bmp;*.dib\0"
+#endif
+#ifdef HAVE_JPEG
                       "JPEG files (*.jpeg;*.jpg;*.jpe)\0*.jpeg;*.jpg;*.jpe\0"
+#endif
+#ifdef HAVE_PNG 
                       "Windows bitmap (*.bmp;*.dib)\0*.bmp;*.dib\0"
```

Comment:
i intended to get results as below.

if HAVE_PNG is true then the list is as below
"PortableNetworkGraphicsfiles(*.png)\0*.png\0"
"JPEGfiles(*.jpeg;*.jpg;*.jpe)\0*.jpeg;*.jpg;*.jpe\0"
"Windowsbitmap(*.bmp;*.dib)\0*.bmp;*.dib\0"
"TIFFFiles(*.tiff;*.tif)\0*.tiff;*.tif\0"
"JPEG-2000files(*.jp2)\0*.jp2\0"
"WebPfiles(*.webp)\0*.webp\0"
"Portableimageformat(*.pbm;*.pgm;*.ppm;*.pxm;*.pnm)\0*.pbm;*.pgm;*.ppm;*.pxm;*.pnm\0"
"OpenEXRImagefiles(*.exr)\0*.exr\0"
"RadianceHDR(*.hdr;*.pic)\0*.hdr;*.pic\0"
"Sunrasterfiles(*.sr;*.ras)\0*.sr;*.ras\0"
"AllFiles(*.*)\0*.*\0";

if HAVE_PNG is false then the list is as below
"Windowsbitmap(*.bmp;*.dib)\0*.bmp;*.dib\0"
"JPEGfiles(*.jpeg;*.jpg;*.jpe)\0*.jpeg;*.jpg;*.jpe\0"
"TIFFFiles(*.tiff;*.tif)\0*.tiff;*.tif\0"
"JPEG-2000files(*.jp2)\0*.jp2\0"
"WebPfiles(*.webp)\0*.webp\0"
"Portableimageformat(*.pbm;*.pgm;*.ppm;*.pxm;*.pnm)\0*.pbm;*.pgm;*.ppm;*.pxm;*.pnm\0"
"OpenEXRImagefiles(*.exr)\0*.exr\0"
"RadianceHDR(*.hdr;*.pic)\0*.hdr;*.pic\0"
"Sunrasterfiles(*.sr;*.ras)\0*.sr;*.ras\0"
"AllFiles(*.*)\0*.*\0";